catapult-sdk/test/plugins/metadata_spec.js

```diff
@@ -0,0 +1,156 @@
+ * along with Catapult.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const EntityType = require('../../src/model/EntityType');
+const ModelSchemaBuilder = require('../../src/model/ModelSchemaBuilder');
+const metadataPlugin = require('../../src/plugins/metadata');
+const test = require('../binaryTestUtils');
+const { expect } = require('chai');
+
+describe('metadata plugin', () => {
+	describe('register schema', () => {
+		it('adds metadata system schema', () => {
+			// Arrange:
+			const builder = new ModelSchemaBuilder();
+			const numDefaultKeys = Object.keys(builder.build()).length;
+
+			// Act:
+			metadataPlugin.registerSchema(builder);
+			const modelSchema = builder.build();
+
+			// Assert:
+			expect(Object.keys(modelSchema).length).to.equal(numDefaultKeys + 3);
+			expect(modelSchema).to.contain.all.keys(['accountMetadata', 'mosaicMetadata', 'namespaceMetadata']);
+
+			// - accountMetadata
+			expect(Object.keys(modelSchema.accountMetadata).length).to.equal(Object.keys(modelSchema.transaction).length + 3);
+			expect(modelSchema.accountMetadata).to.contain.all.keys(['targetPublicKey', 'scopedMetadataKey', 'value']);
+
+			// - mosaicMetadata
+			expect(Object.keys(modelSchema.mosaicMetadata).length).to.equal(Object.keys(modelSchema.transaction).length + 4);
+			expect(modelSchema.mosaicMetadata).to.contain.all.keys(['targetPublicKey', 'scopedMetadataKey', 'targetMosaicId', 'value']);
+
+			// - namespaceMetadata
+			expect(Object.keys(modelSchema.namespaceMetadata).length).to.equal(Object.keys(modelSchema.transaction).length + 4);
+			expect(modelSchema.namespaceMetadata).to.contain.all.keys([
+				'targetPublicKey',
+				'scopedMetadataKey',
+				'targetNamespaceId',
+				'value'
+			]);
+		});
+	});
+
+	describe('register codecs', () => {
+		const getCodecs = () => {
+			const codecs = {};
+			metadataPlugin.registerCodecs({
+				addTransactionSupport: (type, codec) => { codecs[type] = codec; }
+			});
+
+			return codecs;
+		};
+
+		it('adds metadata codecs', () => {
+			// Act:
+			const codecs = getCodecs();
+
+			// Assert: codecs were registered
+			expect(Object.keys(codecs).length).to.equal(3);
+			expect(codecs).to.contain.all.keys([
+				EntityType.accountMetadata.toString(),
+				EntityType.mosaicMetadata.toString(),
+				EntityType.namespaceMetadata.toString()
+			]);
+		});
+
+		const getCodec = entityType => getCodecs()[entityType];
+
+		describe('supports account metadata', () => {
+			const targetPublicKey = test.random.bytes(test.constants.sizes.signer); // 32
+
+			test.binary.test.addAll(getCodec(EntityType.accountMetadata), 60, () => ({
+				buffer: Buffer.concat([
+					Buffer.from(targetPublicKey), // key 32b
+					Buffer.of(0xF2, 0x26, 0x6C, 0x06, 0x40, 0x83, 0xB2, 0x92),
+					Buffer.of(0x03, 0x00),
+					Buffer.of(0x10, 0x00),
+					Buffer.of(0x6d, 0x65, 0x74, 0x61, 0x20, 0x69, 0x6e, 0x66, 0x6f, 0x72, 0x6d, 0x61, 0x74, 0x69, 0x6f, 0x6e)
+				]),
```

Comment:
i'd maybe tag-comment what "field" each buffer relates to (as you did with the // key), same below